src/main/java/com/redhat/pantheon/servlet/BulkDeleteServlet.java

```diff
@@ -64,42 +64,42 @@ public BulkDeleteServlet() {
     
     @Override
     protected void doPost(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) throws ServletException, IOException {
-        String[] checkboxValues = request.getParameterValues(FORM_PARAMETER);
-        if (checkboxValues.length == 0) {
-    		logger.info("No modules selected for delete");
-    		return;
-    	}
-        List<String> resourcePaths = Arrays.asList(checkboxValues);
-        ResourceResolver resourceResolver = request.getResourceResolver();
-        String referrer = request.getHeader("referer");
-        try {
-        	
-        	for ( String rPath: resourcePaths) {
-        		Resource res = resourceResolver.getResource(CONTENT_PATH_PREFIX + rPath);
-        		// Delete the resource.
-        		if (res != null) {
-        			resourceResolver.delete(res);
-        		} else {
-        			String msg = "Missing Resource " + CONTENT_PATH_PREFIX + rPath + " for delete";
-        			response.sendError(HttpServletResponse.SC_NOT_FOUND, msg);
-        			throw new ResourceNotFoundException(msg);
-        		}
-        	}
-		resourceResolver.commit();
-        	response.sendRedirect(referrer);
-        } 
+    	
+    	try {
+    		ResourceResolver resourceResolver = request.getResourceResolver();
+    		String[] checkboxValues = request.getParameterValues(FORM_PARAMETER);
+    		
+            if (checkboxValues != null && checkboxValues.length > 0) {
+            	List<String> resourcePaths = Arrays.asList(checkboxValues);
+            	String referrer = request.getHeader("referer");
+
+            	for ( String rPath: resourcePaths) {
+            		Resource res = resourceResolver.getResource(CONTENT_PATH_PREFIX + rPath);
+            		// Delete the resource.
+            		if (res != null) {
+            			resourceResolver.delete(res);
+            		} else {
+            			String msg = "Missing Resource " + CONTENT_PATH_PREFIX + rPath + " for delete";
+            			response.sendError(HttpServletResponse.SC_NOT_FOUND, msg);
+            			throw new ResourceNotFoundException(msg);
+            		}
+            	}
+            	resourceResolver.commit();
+            	response.sendRedirect(referrer);
+            }
+        }
         catch (Exception e) {
         	// Log the error.
         	logger.error("Module deletion failed: {}", e.getMessage(), e);
 		// Revert all pending changes.
-		if (resourceResolver.hasChanges()) {
-			resourceResolver.revert();
+		if (request.getResourceResolver().hasChanges()) {
+			request.getResourceResolver().revert();
 			String commitError = "Something unexpected happened. Message was: " + e.getMessage();
 			response.sendError(HttpServletResponse.SC_BAD_REQUEST, commitError);
 		}
         } finally {
         	// When done, close the ResourceResolver.
-        	resourceResolver.close();
+        	request.getResourceResolver().close();
```

Comment:
I'm confused why these changes are part of this branch. I went through your commits and I get the impression that this code was throwing exceptions when you ran the python script? If that's so, that's a bit of a concern... I would rather find some mechanism where the delete servlet isn't invoked at all for anything other than a delete operation. However that must have been an existing issue, so for this branch specifically, we'll leave it as-is.

Reply:
bulkDeleteServlet get registered when uploader pushes the same repo over and over. I scheduled a a discussion session on this topic this afternoon.

Reply:
Cool, I saw that. That's going to be a significant change though, so let's create a separate branch if we're going to be reworking the delete servlet.